scripts/bootstrap

```diff
@@ -0,0 +1,200 @@
+        unzip "/home/root/$WGET_BINARIES_FILENAME" -d /home/root/.cache/wget_bin -q
+        rm "/home/root/$WGET_BINARIES_FILENAME"
+
+        cat > /home/root/.cache/wget_bin/wget <<EOF
+#!/bin/sh
+LD_LIBRARY_PATH="/home/root/.cache/wget_bin/dist" /home/root/.cache/wget_bin/dist/wget \$@
+EOF
+
+        chmod +x /home/root/.cache/wget_bin/wget
+    fi
+
+    # Ensure this binary is used
+    if [[ $(command -v wget) != /home/root/.cache/wget_bin/wget ]]; then
+        PATH="/home/root/.cache/wget_bin:$PATH"
+    fi
+}
+
+main() {
+    # Allow changing the default branch
+    [[ -n "$TOLTEC_BRANCH" ]] || TOLTEC_BRANCH="stable"
+
+    # Select source for remarkable_entware repo
+    if [ -z "$REMARKABLE_ENTWARE_REPO_AUTHOR" ]; then
+        # Use LinusCDE's fork for the time beeing
+        # TODO: Switch to a more official repo when it exists
+        REMARKABLE_ENTWARE_REPO_AUTHOR=LinusCDE
+    fi
+
+    # Check status of installation and ensure installed
+    if [ -d /home/root/.entware ]; then
+        if [[ -d /opt ]] && files="$(ls -qA -- /opt)" && [[ -n $files ]]; then
+            # Entware is installed and active
+            log INFO "Entware is installed and active"
+        else
+            # Installed but not active
+            log INFO "Entware is installed but not reenabled"
+            entware_reenable
+        fi
+    else # = not found: /home/root/.entware
+        # Not installed at all
+        log INFO "Entware is not installed"
+        entware_install
+    fi
+
+    # Ensure opkg is in PATH
+    ensure_opkg_available
+    # Ensure wget is installed as opkg package
+    ensure_opt_wget_and_ca_certs_installed
+    # Add toltec repo if mising
+    ensure_toltec_repo_added
+
+    if [ $# -ge 1 ]; then
+        install_packages "$@" || throw "Failed to install requested packages"
+    fi
+}
+
+entware_install() {
+    bootstrap_wget || throw "bootstrap_wget() failed!"
+    (wget -O- "https://raw.githubusercontent.com/$REMARKABLE_ENTWARE_REPO_AUTHOR/remarkable_entware/master/entware_install.sh" | sh) \
+        || throw "Failed to download and/or install remarkable_entware."
+    log INFO "Entware was installed successfully."
+}
+
+entware_reenable() {
+    bootstrap_wget || throw "bootstrap_wget() failed!"
+    (wget -O- "https://raw.githubusercontent.com/$REMARKABLE_ENTWARE_REPO_AUTHOR/remarkable_entware/master/entware_reenable.sh" | sh) \
+        || throw "Failed to download and/or install remarkable_entware."
+    log INFO "Entware was reenabled successfully."
+}
+
+ensure_opkg_available_from_bashrc() {
+    # Run by ensure_opkg_available
+    if ! grep '.*PATH=.*/opt/bin.*' /home/root/.bashrc | grep -v '^#' >/dev/null; then
+        # Path is not in .bashrc
+        echo -e '\n# Path added by bootstrap.sh' >> /home/root/.bashrc
+        echo 'PATH="/opt/bin:/opt/sbin:$PATH"' >> /home/root/.bashrc
+        log WARN '/opt/bin and /opt/sbin were not added to $PATH in ~/.bashrc. Fixed automatically.'
```

Comment:
Isn’t this redundant with `entware_install.sh`?

Reply:
I don't think so, but I'm not sure. `entware_install.sh` may add `/opt/bin:/opt/sbin` temporarily (not sure on that) and just tells the user how to add the path permanently. The part here does that for the user if opkg was installed automatically or the user forgot it.

Reply:
Okay, so this is related to the thread above about merging with `remarkable_entware`.